src/vs/workbench/browser/parts/editor/editorParts.ts

```diff
@@ -635,9 +630,43 @@ export class EditorParts extends MultiWindowParts<EditorPart> implements IEditor
 		return this.getPart(container).createEditorDropTarget(container, delegate);
 	}
 
+	//#endregion
+
+	//#region Editor Group Context Key Handling
+
 	private readonly globalContextKeys = new Map<string, IContextKey<ContextKeyValue>>();
 	private readonly scopedContextKeys = new Map<GroupIdentifier, Map<string, IContextKey<ContextKeyValue>>>();
 
+	private registerContextKeyListeners(): void {
+		this._register(this.onDidChangeActiveGroup(() => this.updateGlobalContextKeys()));
+		this._register(this.onDidAddGroup(group => this.registerRunGroupContextKeyProviders(group)));
+		this._register(this.onDidRemoveGroup(group => {
+			this.removeGroupScopedContextKeys(group);
+			this.unregisterRunGroupContextKeyProviders(group);
+		}));
+	}
+
+	private updateGlobalContextKeys(): void {
+		const activeGroupScopedContextKeys = this.scopedContextKeys.get(this.activeGroup.id);
+		if (!activeGroupScopedContextKeys) {
+			return;
```

Comment:
Do we still have to reset the global context keys if there is no group scoped context keys?

Reply:
No there always is a group.